src/projects/shared/site/src/containers/HistoryBox/index.tsx

```diff
@@ -31,64 +60,69 @@ const HistoryEntry = ({ a, isRedo }: HistoryEntryProps) => {
-//                              onClick={(e) => {
-//                                  // Necessary to stop child entries from displaying
-//                                  //  extra info about the parent history entry
-//                                  e.stopPropagation();
-//                                  setDisplayExtraInfo(!displayExtraInfo);
-//                              }} />
-//                     )}
-//                     <span>{g.getName()}</span>
-//                 </div>
-//                 <span className={`${isCollapsed ? "collapsed" : "" }`}>&rsaquo;</span>
-//             </div>
-//             {displayExtraInfo && g.getCustomInfo?.()?.map((obj, i) =>
-//                 <div key={`group-action-extrainfo-${i}`} className="historybox__groupentry__extrainfo">{obj}</div>
-//             )}
-//             {!isCollapsed && g.getActions().map((a, i) => (
-//                 <HistoryEntry key={`group-action-entry-${i}`}
-//                               a={a}
-//                               isRedo={isRedo} />
-//             ))}
-//             {!isCollapsed && g.isEmpty() && <div style={{ marginLeft: "10px" }}>Empty</div>}
-//         </div>
-//     );
-// }
+type GroupActionEntryProps = {
+    a: LogEntry;
+    isRedo: boolean;
+}
+const GroupActionEntry = ({ a, isRedo }: GroupActionEntryProps) => {
+    const [isCollapsed, setIsCollapsed] = useState(true);
+    // TODO[model_refactor_api]: Any extraInfo we would want to display on the log?
+    const [displayExtraInfo, setDisplayExtraInfo] = useState(false);
+
+    // TODO[model_refactor_api]: Decide if we want to nest single op entries or not
+    // if (a.ops.length === 1) {
+    //     return <HistoryEntry op={a.ops[0]} isRedo={isRedo} />
+    // }
+
+    return (
+        <div className={`historybox__groupentry ${isRedo ? "historybox__groupentry--dashed" : ""}`}
+             role="button" tabIndex={0}
+             onClick={(e) => {
+                 // Necessary to stop child entries from collapsing the parent history entry
+                 e.stopPropagation();
+                 setIsCollapsed(!isCollapsed);
+             }}>
+            <div className="historybox__groupentry__header">
+                <div>
+                    {/* {g.getCustomInfo() && (
+                        <img src="img/icons/info.svg"
+                             alt="Display extra info"
+                             onClick={(e) => {
+                                 // Necessary to stop child entries from displaying
+                                 //  extra info about the parent history entry
+                                 e.stopPropagation();
+                                 setDisplayExtraInfo(!displayExtraInfo);
+                             }} />
+                    )} */}
+                    {/* TODO[model_refactor_api]: clientData isn't used/set now,
+                        should Circuit's commitTransaction take it in as a parameter like CircuitDocument? */}
```

Comment:
Yes, but we need to make a separate entry in LogEntry like `isUndo` or something to track undo operations internally

Reply:
I'll leave this TODO for this pr, but I adjusted the wording on the comment to mention that it is used for undo/redo tracking